src/tardev-snapshotter/tarindex/src/lib.rs

```diff
@@ -427,13 +424,24 @@ pub fn append_index(data: &mut (impl io::Read + io::Write + io::Seek)) -> io::Re
     // Calculate the offsets for directory entries.
     let inode_table_size: u64 = mem::size_of::<Inode>() as u64 * ino_count;
     let string_table_offset = init_direntry_offset(root.clone(), contents_size + inode_table_size)?;
+    let mut symlink_offset = string_table_offset;
 
     // Write the i-node table.
     visit_breadth_first_mut(root.clone(), |e| {
         if e.emitted {
             return Ok(());
         }
 
+        // Check for special symlink names
+        let inode_offset = if (e.mode & S_IFMT) == S_IFLNK &&  e.offset == 0 {
+            symlink_offset
+        } else {
+            e.offset
+        };
+        if (e.mode & S_IFMT) == S_IFLNK &&  e.offset == 0 {
+            symlink_offset += e.size;
```

Comment:
To avoid having to repeat this check, I'd do this on line 437 above.

Something like:
```
    let inode_offset = if (e.mode & S_IFMT) == S_IFLNK && e.offset == 0 {
            let v = symlink_offset;
            symlink_offset += e.size;
            v
        } else {
            e.offset
        }.into();
```

In fact, it's preferable to leave the short end at the top, so I would change to:
```
    let inode_offset = if (e.mode & S_IFMT) != S_IFLNK || e.offset != 0 {
            e.offset
        } else {
            let v = symlink_offset;
            symlink_offset += e.size;
            v
        }.into();
```

Reply:
Thanks for the suggestion! Updated with style of the second block.